File: docs/www/4_examples/progs/src/tutorial/graph_1.f
C (C) Copyright 1996-2016 ECMWF.
C 
C This software is licensed under the terms of the Apache Licence Version 2.0
C which can be obtained at http://www.apache.org/licenses/LICENSE-2.0. 
C In applying this licence, ECMWF does not waive the privileges and immunities 
C granted to it by virtue of its status as an intergovernmental organisation nor
C does it submit to any jurisdiction.
C

      program graf01

      dimension x(11),y(11)
      data x /0.0,1.,2.,3.,4.,5.,6.,7.,8.,9.,10./
      data y /0.0,4.,6.,8.,10.,15.,20.,30.,20.,10.,5./
c
c     open magics and define printer
c
        call popen
        call psetc ('ps_device',    'ps_a4')
        call psetc ('ps_file_name', 'graph_1.ps')

        call psetc ('page_id_line_user_text','My first graph')

      call psetc ('legend','on')
      call psetc ('legend_user_text','My red curve')

c
c	set up axis system
c
      call psetc ('subpage_map_projection','none')  
      call psetc ('axis_orientation','horizontal')
      call psetc ('axis_position','bottom')
      call psetr ('axis_min_value',0.0)
      call psetr ('axis_max_value',10.0)
      call psetr ('axis_tick_interval',1.0)
      call paxis
      call psetc ('axis_orientation','vertical')
      call psetc ('axis_position','left')
      call psetr ('axis_min_value',0.0)
      call psetr ('axis_max_value',40.0)
      call psetr ('axis_tick_interval',5.0)
      call paxis
      
c
c	plot simple curve with symbols
c

c
c	set graph parameters
c
      call psetc ('graph_type','curve')
      call psetc ('graph_symbol','on')
      call pseti ('graph_symbol_marker_index',18)
      call psetc ('graph_line_colour','red')
      call pset1r ('graph_curve_x_values',x,11)
      call pset1r ('graph_curve_y_values',y,11)
      call pgraph
c
c	plot text
c
      call psetc ('text_line_1','Graph plotting in ')
      call psetc ('text_line_2','curve mode with symbols')
      call psetc ('text_line_colour','navy')
      call pseti ('text_line_count',2)
      call ptext
      call psetc ('graph_symbol','off')
      call pclose

      end

File: test/python/graph1.py
# (C) Copyright 1996-2016 ECMWF.
# 
# This software is licensed under the terms of the Apache Licence Version 2.0
# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0. 
# In applying this licence, ECMWF does not waive the privileges and immunities 
# granted to it by virtue of its status as an intergovernmental organisation nor
# does it submit to any jurisdiction.


#importing Magics module
from Magics.macro import *


ref = 'graph1'
#Setting of the output file name
output = output(output_formats = ['png'], 
		output_name_first_page_number = "off",
		output_name = ref)

#Setting the cartesian view
projection = mmap(subpage_map_projection = 'cartesian',
			subpage_x_axis_type = 'regular',
			subpage_y_axis_type = 'regular',
			subpage_x_min = -20.,
			subpage_x_max = 20.,
			subpage_y_min = 25.,
			subpage_y_max = 75.)

#Vertical axis
vertical = maxis(axis_orientation = "vertical",
				 axis_grid =  "on",
				 axis_grid_colour = "grey",
				 axis_grid_thickness = 1,
				 axis_grid_line_style = "dot")

#Horizontal axis
horizontal = maxis(axis_orientation = "horizontal",
				 axis_grid =  "on",
				 axis_grid_colour = "grey",
				 axis_grid_thickness = 1,
				 axis_grid_line_style = "dot")



#define  the  data 
x = numpy.array([-15.,0.,15.])
y = numpy.array([50.,30.,50.])

#define the input data
input = minput(input_x_values =  x,
			input_y_values =  y)

#Define the graph 
graph = mgraph( graph_line_colour  = "red",
            graph_line_thickness = 8,
            graph_symbol = "on",
            legend =  "on",
            legend_user_text =  "<font colour='red'> My red curve </font>",
            graph_symbol_marker_index =  15,
            graph_symbol_colour =  "black",
            graph_symbol_height =  1.)




title = mtext(
           text_lines = ["Simple Graph with legend"],
		   text_justification = "left",
		   text_font_size = 1.,
           text_colour =  "charcoal")


#To the plot
plot(output, projection, vertical, horizontal, input, graph, title)

#For documentation only
tofortran(ref, output, projection, vertical, horizontal, input, graph, title)
tomv4(ref, graph)
tohtml(ref, input,  graph)














